### PR TITLE
EDSC-3819: Fixing `variables` object  in submitHarmonyOrder

### PR DESCRIPTION
**Pull request by daniel-zamora**
# Overview
Bug fix for Harmony based orders handling variables
### What is the feature?
We add variables to the permittedAccessMethodFields so that it is available for the build url function farther down
### What is the Solution?
We add variables to the permittedAccessMethodFields so that it is available for the build url function farther down
### What areas of the application does this impact?
Harmony Orders completing correctly
# Testing
The ticket includes example orders in uat and prod that can be re-ordered for testing.
# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/util/retrievals.js [hunk 1]
```diff
@@ -108,12 +108,31 @@ export const prepareRetrievalParams = (state) => {
 
     const params = buildGranuleSearchParams(preparedParams)
 
+    const allVariables = accessMethods[selectedAccessMethod].variables
```
Comment (macrouch) [inline static/src/js/util/retrievals.js:111]: I'm surprised eslint let this go through, you should destructure variables off of the object

File: static/src/js/util/retrievals.js [hunk 2]
```diff
@@ -108,12 +108,31 @@ export const prepareRetrievalParams = (state) => {
...
+
     returnValue.granule_params = params
     returnValue.access_method = pick(
       accessMethods[selectedAccessMethod],
       permittedAccessMethodFields
     )
 
+    if (returnValue.access_method.selectedVariables) {
```
Comment (macrouch) [inline static/src/js/util/retrievals.js:119]: avoid chaining multiple calls into nested objects like this, and below in the loop

File: static/src/js/util/retrievals.js [hunk 3]
```diff
@@ -108,12 +108,31 @@ export const prepareRetrievalParams = (state) => {
...
+      const variableObjects = []
+      returnValue.access_method.selectedVariables.forEach((variableKey) => {
```
Comment (macrouch) [inline static/src/js/util/retrievals.js:121]: This looks like you want to be doing a `.map()`, not a `.forEach()`

**Review by Bccorb — APPROVED** [review]

**Review by rushgeo — APPROVED** [review]
Looks good to me. For the example customized download for ATL03 in the issue, I got a correctly subsetted HDF5 with just the single variable requested.

**Pull request merged**